File: backend/src/domain/note.py
import datetime

from src.domain.meeting_id import MeetingId


class Note:
    def __init__(
        self,
        meeting_id: MeetingId,
        title: str,
        content: str,
        markdown: str,
        created_at: datetime.datetime = datetime.datetime.now(),
    ) -> None:
        self._meeting_id = meeting_id
        self._title = title
        self._content = content
        self._markdown = markdown
        self._created_at = created_at

    @property
    def id(self) -> MeetingId:
        return self._meeting_id

    @property
    def title(self) -> str:
        return self._title

    @property
    def text(self) -> str:
        return self._content

    @property
    def markdown(self) -> str:
        return self._markdown

    @property
    def created_at(self) -> datetime.datetime:
        return self._created_at

    def __dict__(self) -> dict:
        return {
            "meeting_id": self._meeting_id.value,
            "title": self._title,
            "content": self._content,
            "markdown": self._markdown,
            "created_at": self._created_at,
        }

    def __str__(self) -> str:
        return f"{self.title}: {self.text}"

File: frontend/src/components/Note.js
import {Link} from "react-router-dom";
import { format } from 'date-fns';

export const truncate = (text, length = 100) => {
  if (text.length > length) {
    return text.substring(0, length) + '...';  // Appends ellipsis to indicate truncation
  }
  return text;
};


export function Note({meeting_id, title, created_at, content}) {
  console.log(title);
  return (
    <Link
      to={`/note/${meeting_id}`}
      className="block w-full p-6 bg-our_dark outline rounded-lg shadow hover:outline-4 hover:outline-our_magenta min-h-60 shadow-md hover:shadow-xl"
    >
      <h5 className="mb-2 text-2xl font-bold tracking-tight text-white">
        {title.replaceAll("\"", "")}
      </h5>
      <p className="font-normal text-gray-200 mb-2">{format(new Date(created_at), 'dd.MM.yyyy')}</p>
      <p className="font-normal text-gray-400 items-end">{truncate(content, 200)}</p>
    </Link>
  );
}
